### PR TITLE
fix torch engine infer

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Fix bug in torch engine infer method

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/pytorch/engine/engine.py [hunk 1]
```diff
@@ -690,11 +690,8 @@ def batched_infer(self,
         """Send inference request.
 
         Args:
-            session_id (int): The session id.
-            prompt_token_ids (List[int]): The input token ids.
-            request_output_len (int): The max output length of this request.
-            step (int): No use for now.
-            sampling_param (SamplingParam): The sampling param of the output.
+            session_ids (List[int]): The session id.
```
Comment (lvhan028) [inline lmdeploy/pytorch/engine/engine.py:693]: miss `gen_config`

File: lmdeploy/pytorch/engine/engine.py [hunk 2]
```diff
@@ -1016,13 +1010,13 @@ def infer(self,
...
         for outputs in self.stream_infer(session_id,
-                                         prompt_token_ids,
+                                         input_ids,
                                          gen_config=gen_config,
                                          **kwargs):
             status, tmp_ids, _ = outputs
             if status not in [ResponseType.SUCCESS, ResponseType.FINISH]:
                 return (status, token_ids, len(token_ids))
-            token_ids += tmp_ids
+            token_ids = tmp_ids
```
Comment (lvhan028) [inline lmdeploy/pytorch/engine/engine.py:1019]: We don't need to append `tmp_id`?

Comment (RunningLeon) [reply lmdeploy/pytorch/engine/engine.py:1019]: the output `token_ids` from `stream_infer` is the whole token_ids as accumulated in here
https://github.com/InternLM/lmdeploy/blob/fad7c90ffd206eddd84c2336742c4e6a66c64834/lmdeploy/pytorch/engine/engine.py#L988

**Review by lvhan028 — APPROVED** [review]
LGTM

**Review by grimoire — APPROVED** [review]

**Pull request merged**